plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,212 @@
+        } else {
+          process.stdout.write(newContents);
+        }
+      });
+    });
+
+/**
+ * Gets the database of renames.
+ * @return {!Promise<Object>} The database of renames as an object.
+ */
+export async function getDatabase() {
+  const response = await fetch(DATABASE_URL);
+  const body = await response.text();
+  return JSON5.parse(body);
+}
+
+/**
+ * Upgrades pieces of the Blockly API that have been renamed by modifying
+ * developers' local files.
+ * @param {!Object} database The database of renames.
+ * @param {string} currVersion The version to migrate from.
+ * @param {string} newVersion The version to migrate to.
+ * @param {!Array<string>} strings The strings to apply the renamings in.
+ * @return {!Array<string>} The strings with renamings applied.
+ * @deprecated Use Renamer instance instead.
+ */
+export function doRenamings(database, currVersion, newVersion, strings) {
+  const renamer = new Renamer(database, currVersion, newVersion);
+  return strings.map((str) => renamer.rename(str));
+}
+
+/**
+ * An object which can apply a given set of renamings to the contents
+ * of JavaScript files, so as to automatically update developer's code
+ * where parts of the Blockly API have been renamed.
+ */
+export class Renamer {
+  /**
+   * Constructor for the class.
+   * @param {!Object} database The database of renames.
+   * @param {string} currVersion The version to migrate from.
+   * @param {string} newVersion The version to migrate to.
+   */
+  constructor(database, currVersion, newVersion) {
+    /** @private @const {!Array<!VersionRenamer>} */
+    this.versionRenamers_ =
+        Renamer.calculateRenamings(database, currVersion, newVersion);
+  }
+
+  /**
+   * Turns a database of renamings into an intermediate format that is
+   * easier to apply.
+   * @param {!Object} database The database of renames.
+   * @param {string} currVersion The version to migrate from.
+   * @param {string} newVersion The version to migrate to.
+   * @return {!Array<!VersionRenamer>} The collection of renamings to perform.
+   */
+  static calculateRenamings(database, currVersion, newVersion) {
+    currVersion = versionUtils.coerce(currVersion);
+    newVersion = versionUtils.coerce(newVersion);
+    const versions = Object.keys(database).sort(versionUtils.compare);
+    const renamers /** !Array<!VersionRenamer> */ = [];
+    for (const version of versions) {
+      // Only process versions in the range (currVersion, ^newVersion].
+      if (versionUtils.lte(version, currVersion)) continue;
+      if (versionUtils.gt(version, newVersion)) break;
+
+      renamers.push(new VersionRenamer(database[version]));
+    }
+    return renamers;
+  }
+
+  /**
+   * Applies the given renamings directly to a JavaScript string
+   * (presumably the contents of a developer's file).
+   * @param {string} str The string to apply the renamings in.
+   * @return {string} The file with renamings applied.
```

Comment:
nit: to be consistent with the other language here, you're returning a string, not a file